X-Mozilla-Status: 0001
X-Mozilla-Status2: 00000000
Return-Path: <linux-omap-owner@vger.kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.2.2 (2007-07-23) on
	morningsun.geekisp.com
X-Spam-Level: 
X-Spam-Status: No, score=-3.0 required=5.0 tests=AWL,DKIM_POLICY_SIGNSOME,
	DK_POLICY_SIGNSOME,RCVD_IN_DNSWL_MED autolearn=ham version=3.2.2
Delivered-To: balister.org-philip@balister.org
Received: (qmail 21713 invoked by uid 1003); 18 Jul 2008 01:35:29 -0000
Received: from vger.kernel.org (209.132.176.167)
  by mail.geekisp.com with SMTP; 18 Jul 2008 01:35:29 -0000
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
	id S1754758AbYGRBf2 (ORCPT <rfc822;philip@balister.org>);
	Thu, 17 Jul 2008 21:35:28 -0400
Received: (majordomo@vger.kernel.org) by vger.kernel.org id S1757008AbYGRBf1
	(ORCPT <rfc822;linux-omap-outgoing>);
	Thu, 17 Jul 2008 21:35:27 -0400
Received: from utopia.booyaka.com ([72.9.107.138]:49364 "EHLO
	utopia.booyaka.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
	with ESMTP id S1754758AbYGRBfY (ORCPT
	<rfc822;linux-omap@vger.kernel.org>); Thu, 17 Jul 2008 21:35:24 -0400
Received: (qmail 13880 invoked by uid 526); 18 Jul 2008 01:35:23 -0000
MBOX-Line: From nobody Thu Jul 17 19:34:51 2008
From:	Paul Walmsley <paul@pwsan.com>
Subject: [PATCH 4/9] TWL4030: clear MADC interrupt status registers upon init
To:	linux-omap@vger.kernel.org
Date:	Thu, 17 Jul 2008 19:34:51 -0600
Message-ID: <20080718013450.18943.33397.stgit@localhost.localdomain>
In-Reply-To: <20080718013205.18943.34047.stgit@localhost.localdomain>
References: <20080718013205.18943.34047.stgit@localhost.localdomain>
User-Agent: StGIT/0.14.3.163.g06f9
MIME-Version: 1.0
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 7bit
Sender:	linux-omap-owner@vger.kernel.org
Precedence: bulk
List-ID: <linux-omap.vger.kernel.org>
X-Mailing-List:	linux-omap@vger.kernel.org

twl_init_irq() does not clear MADC interrupt status registers upon init -
fix.

Signed-off-by: Paul Walmsley <paul@pwsan.com>
---

 drivers/i2c/chips/twl4030-core.c |   14 ++++++++++++++
 1 files changed, 14 insertions(+), 0 deletions(-)

diff --git a/drivers/i2c/chips/twl4030-core.c b/drivers/i2c/chips/twl4030-core.c
index bb0732c..9d93524 100644
--- a/drivers/i2c/chips/twl4030-core.c
+++ b/drivers/i2c/chips/twl4030-core.c
@@ -821,6 +821,20 @@ static void twl_init_irq(void)
 		return;
 	}
 
+	/* MADC_ISR1 */
+	res = twl4030_i2c_write_u8(TWL4030_MODULE_MADC, 0xFF, 0x61);
+	if (res < 0) {
+		pr_err("%s[%d][%d]\n", msg, res, __LINE__);
+		return;
+	}
+
+	/* MADC_ISR2 */
+	res = twl4030_i2c_write_u8(TWL4030_MODULE_MADC, 0xFF, 0x63);
+	if (res < 0) {
+		pr_err("%s[%d][%d]\n", msg, res, __LINE__);
+		return;
+	}
+
 	/* key Pad */
 	/* KEYPAD - IMR1 */
 	res = twl4030_i2c_write_u8(TWL4030_MODULE_KEYPAD, 0xFF, (0x12));


--
To unsubscribe from this list: send the line "unsubscribe linux-omap" in
the body of a message to majordomo@vger.kernel.org
More majordomo info at  http://vger.kernel.org/majordomo-info.html

